presto-main/src/main/java/com/facebook/presto/execution/SqlTaskManager.java

```diff
@@ -153,6 +161,7 @@ public SqlTaskManager(
                         nodeInfo.getNodeId(),
                         queryContexts.getUnchecked(taskId.getQueryId()),
                         sqlTaskExecutionFactory,
+                        new ExchangeClientManager(exchangeClientSupplier),
```

Comment:
nit: i would pass the `exchangeClientSupplier` to the `SqlTask`, and create the `ExchangeClientManager` inside the `SqlTask`, as it always has to be task scope.

Reply:
All the tasks has to share the same `ExchangeClientManager`, so the `ExchangeClientManager` have a list of al the exchange clients (via `getExchangeClients`)

Reply:
hmm... I think `ExchangeClient` also need to be removed from `ExchangeClientManager`